File: app/lib/models/meme.dart
class Meme {
  final String id;
  final String title;
  final String source;
  final String? url;
  final String? summary;
  final String? tags;
  final bool kept;
  final bool discarded;
  final bool asked;
  final DateTime fetchedAt;

  const Meme({
    required this.id,
    required this.title,
    this.source = 'bilibili',
    this.url,
    this.summary,
    this.tags,
    this.kept = false,
    this.discarded = false,
    this.asked = false,
    required this.fetchedAt,
  });

  factory Meme.fromJson(Map<String, dynamic> json) {
    return Meme(
      id: json['id'] as String,
      title: json['title'] as String,
      source: (json['source'] as String?) ?? 'bilibili',
      url: json['url'] as String?,
      summary: json['summary'] as String?,
      tags: json['tags'] as String?,
      kept: (json['kept'] as bool?) ?? false,
      discarded: (json['discarded'] as bool?) ?? false,
      asked: (json['asked'] as bool?) ?? false,
      fetchedAt: DateTime.parse(json['fetched_at'] as String),
    );
  }
}

File: backend/app/models/meme.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.database import Base


class Meme(Base):
    """阿玖每天抓取、筛选后准备注入闲聊的热梗。"""

    __tablename__ = "memes"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    source: Mapped[str] = mapped_column(String(50), default="bilibili", nullable=False)
    url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    tags: Mapped[str | None] = mapped_column(String(500), nullable=True)

    kept: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    discarded: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    asked: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)

    fetched_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
